### PR TITLE
UI

### PR DESCRIPTION
**Pull request by IamYunusAli**
in this pull request, I @IamYunusAli have managed to do the following things.

- Create an interactive console app based on the following template
    - [UI example](https://www.youtube.com/watch?v=vkkgrhD6aXQ)
- Create essential functions in `app.rb` that have the following functionalities 
  - List all books.
  - List all people.
  - Create a person (teacher or student, not a plain `Person`).
  - Create a book.
  - Create a rental.
  - List all rentals for a given person's id.
- In `main.rb`  included the following things
  - Present the user with a list of options to perform.
  - Lets users choose an option.
  - If needed, ask for parameters for the option.
  - Have a way to quit the app.

will love to hear more exciting comments ... thank you for reviewing my project.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    puts "Input Date | today date is: #{Date.today}"
+    date = gets.chomp
+    rental = Rental.new(date, books[book_index], people[person_index])
+    @rentals.push(rental)
+    puts 'Rental created successfully'
+  end
+
+  def list_rentals_by_id
+    @rentals.each { |x| puts "Title: #{x.book.title} Author: #{x.book.author} Date: #{x.date} Person: #{x.person.id}" }
+    print 'ID of person: '
```
Comment (HAMMAS-SALEEM) [inline app.rb]: - [ ] When we do use the last option which is list the rentals by id, so when we use it first it should show the **person id** and **person name** and that's all at first and nothing should be shown before giving any id for the input 😊 The reason is you are not providing the **person name** and we cannot memorize the id of each person 😄

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

## Hey @IamYunusAli :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following some requirements

Good job on implementing some of the requirements, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 1 above)
Comment (IamYunusAli) [reply app.rb]: My first idea was like what you said did include the books to make sure of thing and forgot to change it thank you resolved.

**Review by Bondok6 — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

## Hi @IamYunusAli 🙋‍♂️, 
Your project is complete!  There is nothing else to say other than... Great job 👏 
Congratulations! 🎉 

TO HIGHLIGHT:

- Your project is professional ✅
- Your repo is professional ✅
- Well structured files ✅
- Professional README.md file 💯


Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 




## Optional suggestions

**- [OPTIONAL] I noticed that the MIT Link is broken in your README file, so I suggest fixing it 
by changing the path from `[MIT](./LICENSE)` to `[MIT](./MIT.md)` 💯👍**

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**